src/antidote_db.erl

```diff
@@ -0,0 +1,133 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2014 SyncFree Consortium.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(antidote_db).
+
+-include_lib("antidote_utils/include/antidote_utils.hrl").
+
+-export([
+    new/2,
+    close_and_destroy/2,
+    close/1,
+    get_snapshot/3,
+    put_snapshot/4,
+    get_ops/4,
+    put_op/4]).
+
+-type antidote_db() :: {leveldb, eleveldb:db_ref()}.
+
+-type antidote_db_type() :: leveldb.
+
+-export_type([antidote_db/0, antidote_db_type/0]).
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+%% Given a name, returns a new AntidoteDB (for now, only ElevelDB is supported)
+%% OpenOptions are set to use Antidote special comparator in the case of Eleveldb
+-spec new(atom(), antidote_db_type()) -> {ok, antidote_db()} | {error, any()}.
+new(Name, Type) ->
+    case Type of
+        leveldb ->
+            {ok, Ref} = eleveldb:open(Name, [{create_if_missing, true}, {antidote, true}]),
+            {ok, {leveldb, Ref}};
+        _ ->
+            {error, type_not_supported}
+    end.
+
+
+%% Closes and destroys the given base
+-spec close_and_destroy(antidote_db(), atom()) -> ok | {error, any()}.
+close_and_destroy({Type, DB}, Name) ->
+    case Type of
+        leveldb ->
+            eleveldb:close(DB),
+            eleveldb:destroy(Name, []);
+        _ ->
+            {error, type_not_supported}
+    end.
+
+-spec close(antidote_db()) -> ok | {error, any()}.
+close({Type, DB}) ->
+    case Type of
+        leveldb ->
+            eleveldb:close(DB);
+        _ ->
+            {error, type_not_supported}
+    end.
+
+%% Gets the most suitable snapshot for Key that has been committed
+%% before CommitTime. If its nothing is found, returns {error, not_found}
+-spec get_snapshot(antidote_db:antidote_db(), key(),
+    snapshot_time()) -> {ok, snapshot(), snapshot_time()} | {error, not_found}.
+get_snapshot({Type, DB}, Key, CommitTime) ->
```

Comment:
can you use a record for the snapshot please?
Something like here:
https://github.com/SyncFree/antidote/blob/master/include/antidote.hrl#L141
Except I guess also include the "snapshot_time" as part of the record as well